bindings/http/http.go

```diff
@@ -99,6 +102,13 @@ func (h *HTTPSource) Invoke(ctx context.Context, req *bindings.InvokeRequest) (*
 				return nil, fmt.Errorf("invalid path: %s", path)
 			}
 		}
+
+		if treatNon200AsErrorStr, ok := req.Metadata["treatNon200AsError"]; ok {
```

Comment:
I am not a big fan of this property name, as it’s long and it’s technically not just 200, but 2xx status codes.
Can you suggest another name?

Reply:
What about this being a metadata of the component and not of the request? Or, even better, allow both. I predict most people want this behavior to be consistent and not per request.

Reply:
ErrorIfNot2XX

Reply:
I would advocate both, changing just the component setting could relate to wider reaching changes. I can change the name to `errorIfNot2XX` both for the request and component metadata.